stageleft/src/type_name.rs

```diff
@@ -11,14 +11,106 @@ struct RewriteAlloc {
 
 impl VisitMut for RewriteAlloc {
     fn visit_path_mut(&mut self, i: &mut syn::Path) {
-        if i.segments.iter().take(2).collect::<Vec<_>>()
```

Comment:
This... makes me quite sad to be honest. I think in near-term we will need a way for users to provide such redirects for private types re-exported as public, and then in the longer-term we really need to tap into rustc's builtin logic which recommend users the publicly exported path.